code/datums/medal_awards.dm

```diff
@@ -160,7 +161,102 @@ GLOBAL_LIST_INIT(human_medals, list(MARINE_CONDUCT_MEDAL, MARINE_BRONZE_HEART_ME
 
 	return TRUE
 
-/proc/print_medal(mob/living/carbon/human/user, obj/printer)
+/proc/give_medal_award_prefilled(medal_location, mob/giving_mob, _recipient_name, _recipient_rank, _recipient_ckey, reason, _medal_type)
+	var/list/recipient_ranks = list()
+	for(var/datum/data/record/record in GLOB.data_core.general)
+		var/recipient_name = record.fields["name"]
+		recipient_ranks[recipient_name] = record.fields["rank"]
+
+	var/chosen_recipient = _recipient_name
+	if(!chosen_recipient)
+		return FALSE
+
+	// Pick a medal
+	var/medal_type = _medal_type
+	if(!medal_type)
+		return FALSE
+
+	// Write a citation
+	var/citation = strip_html(reason)
+	if(!citation)
+		return FALSE
+
+	// Get mob information
+	var/recipient_rank = _recipient_rank
+	var/posthumous = TRUE
+	var/recipient_ckey = _recipient_ckey
+	var/mob/recipient_mob
+	var/found_other = FALSE
```

Comment:
why use intermediate variables? you can use the args directly (and wihtout a _)

Reply:
because I copy pasted everything. will change

Reply:
Fixed